v2/gcs-to-sourcedb/src/test/java/com/google/cloud/teleport/v2/templates/GCSToSourceDbWithReaderIT.java

```diff
@@ -0,0 +1,258 @@
+import static org.apache.beam.it.truthmatchers.PipelineAsserts.assertThatPipeline;
+
+import com.google.cloud.spanner.Mutation;
+import com.google.cloud.teleport.metadata.SkipDirectRunnerTest;
+import com.google.cloud.teleport.metadata.TemplateIntegrationTest;
+import com.google.cloud.teleport.v2.spanner.migrations.shard.Shard;
+import com.google.common.io.Resources;
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import org.apache.beam.it.common.PipelineLauncher;
+import org.apache.beam.it.common.PipelineLauncher.LaunchConfig;
+import org.apache.beam.it.common.utils.PipelineUtils;
+import org.apache.beam.it.common.utils.ResourceManagerUtils;
+import org.apache.beam.it.gcp.TemplateTestBase;
+import org.apache.beam.it.gcp.dataflow.FlexTemplateDataflowJobResourceManager;
+import org.apache.beam.it.gcp.spanner.SpannerResourceManager;
+import org.apache.beam.it.gcp.storage.GcsResourceManager;
+import org.apache.beam.it.jdbc.CustomMySQLResourceManager;
+import org.apache.beam.it.jdbc.JDBCResourceManager;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Integration test for {@link GCSToSourceDb} Flex template without launching reader job. */
+@Category({TemplateIntegrationTest.class, SkipDirectRunnerTest.class})
+@TemplateIntegrationTest(GCSToSourceDb.class)
+@RunWith(JUnit4.class)
+public class GCSToSourceDbWithReaderIT extends TemplateTestBase {
+
+  private static final Logger LOG = LoggerFactory.getLogger(GCSToSourceDbWithReaderIT.class);
+
+  private static final String SPANNER_DDL_RESOURCE = "GCSToSourceDbWithReaderIT/spanner-schema.sql";
+  private static final String SESSION_FILE_RESOURSE = "GCSToSourceDbWithReaderIT/session.json";
+
+  private static final String TABLE = "Users";
+  private static HashSet<GCSToSourceDbWithReaderIT> testInstances = new HashSet<>();
+  private static PipelineLauncher.LaunchInfo writerJobInfo;
+  private static PipelineLauncher.LaunchInfo readerJobInfo;
+  public static SpannerResourceManager spannerResourceManager;
+  private static SpannerResourceManager spannerMetadataResourceManager;
+  private static CustomMySQLResourceManager jdbcResourceManager;
+  private static GcsResourceManager gcsResourceManager;
+  private static FlexTemplateDataflowJobResourceManager flexTemplateDataflowJobResourceManager;
+
+  /**
+   * Setup resource managers and Launch dataflow job once during the execution of this test class.
+   *
+   * @throws IOException
+   */
+  @Before
+  public void setUp() throws IOException {
```

Comment:
Saw a comment in one of the PRs to move the common logic of set up to a base class. 
All the ITs can then use this base class.

Reply:
I have added the base class in the next PR with Datatypes IT.